### PR TITLE
Create SURLCreator file

### PR DESCRIPTION
**Pull request by firewings2515**

### CODE REVIEW

File: src/SURLCreator.h [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+  // transfer hex to bits
+  static std::string hex2bits(char hex);
+
+  // the base64 table for mapping id to character
+  static const std::string base64_table;
+};
+
+} // namespace shortlink
+
+#endif // !_SHORTLINK_SRC_SURLCreator_H_
```
Comment (will889889) [inline src/SURLCreator.h:36]: please add new line.

**Review by will889889 — CHANGES_REQUESTED** [review]

File: src/SURLCreator.h (hunk 1 above)
Comment (john0312) [reply src/SURLCreator.h:36]: This is a reasonable review. You can also ask for clang-format.

File: src/SURLCreator.cpp [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+  for (int i = 0; binary.size() - i >= 6; i += 6) {
+    std::string subChar = binary.substr(i, 6);
+    int ids = std::stoi(subChar, 0, 2);
+    surl_key += base64_table[ids];
+  }
+
+  return surl_key;
+}
+std::string SURLCreator::hex2bits(char hex) {
+  switch (hex) {
```
Comment (john0312) [inline src/SURLCreator.cpp:41]: Consider using a const static unordered_map

File: src/SURLCreator.cpp [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+
+  // transfer to binary
+  std::string binary = "";
+  for (int i = 0; i < result.size(); i++) {
+    binary += hex2bits(result[i]);
+  }
+
+  // transfer each six bit to base64 char
+  std::string surl_key = "";
+  for (int i = 0; binary.size() - i >= 6; i += 6) {
```
Comment (john0312) [inline src/SURLCreator.cpp:32]: What about the extra bits?

File: src/SURLCreator.cpp [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+    std::string subChar = binary.substr(i, 6);
+    int ids = std::stoi(subChar, 0, 2);
+    surl_key += base64_table[ids];
+  }
+
+  return surl_key;
```
Comment (john0312) [inline src/SURLCreator.cpp:38]: Would this be too long? How long do you want your short URLs to be?

Comment (firewings2515) [reply src/SURLCreator.cpp:43]: the length of short url is 8.
total length of surl_key is 10.
So we drop out the last two.

**Review by john0312 — APPROVED** [review]
Please add license before merging.

**Pull request merged**